src/averageAges.js

```diff
@@ -55,7 +70,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenAges = people.filter(human => onlyWithSon
+    ? ((human.sex === 'm') === onlyWithSon)
+      && people.find(mom => (human.mother === mom.name))
+    : people.find(mom => human.mother === mom.name)
+  ).map(child => child.born
+    - people.find(mom => child.mother === mom.name).born);
+  const averageAge = womenAges.reduce((a, b) => a + b) / womenAges.length;
```

Comment:
use reduce on all women's array and find age difference inside the reducer function 
no need to create a separate array of ages